iOS: Misc tooling updates

chinmaygarde opened the pull request:
Use the project name to initialize the bundle name and identifier

chinmaygarde committed:
iOS: Treat the initial Info.plist string as a mustache template.
Use the project name to initialize the bundle name and identifier

eseidelGoogle commented:
Does this affect `flutter trace` as well?  `flutter stop`?  Seems a bit scary to land right before the hackathon, but depends on @abarth's plans for the branch.

chinmaygarde commented:
> Does this affect `flutter trace` as well? `flutter stop`?

If they use the ID specified in the IOSApp instance, then yes.

abarth commented on packages/flutter_tools/lib/src/application_package.dart:
```diff
@@ -81,13 +82,25 @@ class AndroidApk extends ApplicationPackage {
+
+  factory IOSApp.fromBuildConfiguration(BuildConfiguration config) {
+    if (getCurrentHostPlatform() != HostPlatform.mac) {
+      return null;
```
Is it legal to return null from a factory?

chinmaygarde replied:
From a factory, it should be legal. It's a tiny bit odd that factories look like constructors in Dart. So I am not sure if this is idiomatic.

chinmaygarde committed:
Rewrite Local.xcconfig per build

In case the user changes the path of the Flutter project on disk, the Dart SDK or the Flutter repository itself, the old Xcode paths would be invalid. Now, we rewrite them when we build the project.

chinmaygarde commented:
I have added another patch to this commit:
In case the user changes the path of the Flutter project on disk, the Dart SDK or the Flutter repository itself, the old Xcode paths would be invalid. Now, we rewrite them when we build the project.

chinmaygarde committed:
Add an issue template for Github

chinmaygarde committed:
Fix flutter_tools tests

abarth commented on ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,14 @@
+
+_Detailed issue description here. More information at [https://flutter.io/bug_reports/](https://flutter.io/bug_reports/)_
```
Did you mean to include this file in this patch?

chinmaygarde replied:
Eh, no. Things got backed up over the past couple of days. Do you want me to separate this into another PR?

abarth replied:
Let's send it separately.

abarth commented:
LGTM
